xilem/src/style.rs

```diff
@@ -0,0 +1,109 @@
+// Copyright 2025 the Xilem Authors
+// SPDX-License-Identifier: Apache-2.0
+
+//! Traits used to set custom styles on elements.
+
+use masonry::core::Property;
+use masonry::properties::types::Gradient;
+use masonry::properties::{Background, BorderColor, BorderWidth, BoxShadow, CornerRadius, Padding};
+use vello::peniko::Color;
+
+use crate::property_tuple::PropertyTuple;
+
+/// Marker trait implement by elements to signal that a given property can be set on them.
+pub trait HasProperty<P: Property> {}
```

Comment:
I feel like the natural implementation would have this trait contain a method to get an exclusive reference to the property's value. It's trivial to write a macro to do that, although the DX is a bit ugly without the index metavariable.